packages/lesswrong/client/autoRefresh.ts

```diff
@@ -0,0 +1,65 @@
+import { onStartup, isClient } from '../lib/executionEnvironment';
+import { disconnectDdp, reconnectDdp } from '../lib/meteorDdp';
+import { parsePath, parseRoute } from '../lib/vulcan-core/appContext';
+import { Reload } from 'meteor/reload';
+
+var disconnectTimer: any = null;
+
+// 60 seconds by default
+var disconnectTime = 60 * 1000;
+
+onStartup(disconnectIfHidden);
+
+document.addEventListener('visibilitychange', disconnectIfHidden);
+
+function disconnectIfHidden() {
+    removeDisconnectTimeout();
+
+    if (document.hidden) {
+      createDisconnectTimeout();
+    } else {
+      reconnectDdp();
+    }
+}
+
+function createDisconnectTimeout() {
+    removeDisconnectTimeout();
+
+    disconnectTimer = setTimeout(function () {
+        disconnectDdp();
+    }, disconnectTime);
+}
+
+function removeDisconnectTimeout() {
+    if (disconnectTimer) {
+        clearTimeout(disconnectTimer);
+    }
+}
+
+function shouldAutoRefresh() {
```

Comment:
Renamed from `disconnect_meteor.ts`. Above this line is preexisting stuff, below here is new.